apps/next/src/memlab/lib/baseline.ts

```diff
@@ -0,0 +1,319 @@
+
+/**
+ * Baseline entry for a component
+ */
+export interface BaselineEntry {
+  /** Component name */
+  component: string;
+  /** Timestamp of baseline capture */
+  timestamp: string;
+  /** Git commit hash when baseline was captured */
+  commitHash?: string;
+  /** Number of leaks in baseline */
+  leakCount: number;
+  /** Total retained size in baseline */
+  totalRetainedSize: number;
+  /** Whether the baseline was passing */
+  passed: boolean;
+}
+
+/**
+ * Full baseline data structure
+ */
+export interface BaselineData {
+  /** Version of baseline format */
+  version: number;
+  /** Last updated timestamp */
+  lastUpdated: string;
+  /** Component baselines */
+  components: Record<string, BaselineEntry>;
+  /** Historical trend data */
+  history: Array<{
+    timestamp: string;
+    commitHash?: string;
+    components: Record<
+      string,
+      { leakCount: number; totalRetainedSize: number }
+    >;
+  }>;
+}
+
+/**
+ * Comparison result between current run and baseline
+ */
+export interface BaselineComparison {
+  /** Component name */
+  component: string;
+  /** Current leak count */
+  currentLeakCount: number;
+  /** Baseline leak count */
+  baselineLeakCount: number;
+  /** Change in leak count */
+  leakCountDelta: number;
+  /** Current retained size */
+  currentRetainedSize: number;
+  /** Baseline retained size */
+  baselineRetainedSize: number;
+  /** Change in retained size */
+  retainedSizeDelta: number;
+  /** Whether this represents a regression */
+  isRegression: boolean;
+  /** Whether this represents an improvement */
+  isImprovement: boolean;
+}
+
+const BASELINE_VERSION = 1;
+const DEFAULT_BASELINE_PATH = '.memlab-baseline.json';
+
+/**
+ * Load baseline data from file
+ */
+export function loadBaseline(baselinePath?: string): BaselineData | null {
+  const filePath = baselinePath || DEFAULT_BASELINE_PATH;
+
+  if (!fs.existsSync(filePath)) {
+    return null;
+  }
+
+  try {
+    const content = fs.readFileSync(filePath, 'utf-8');
+    return JSON.parse(content);
+  } catch {
+    console.warn(`⚠️ Failed to load baseline from ${filePath}`);
+    return null;
+  }
+}
+
+/**
+ * Save baseline data to file
+ */
+export function saveBaseline(data: BaselineData, baselinePath?: string): void {
+  const filePath = baselinePath || DEFAULT_BASELINE_PATH;
+
+  try {
+    fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
+    console.log(`📊 Baseline saved to ${filePath}`);
+  } catch (error) {
+    console.error(`❌ Failed to save baseline: ${error}`);
+  }
+}
+
+/**
+ * Update baseline with new results
+ */
+export function updateBaseline(
+  report: MemlabReport,
+  existingBaseline?: BaselineData | null,
+  commitHash?: string,
+): BaselineData {
+  const now = new Date().toISOString();
+
+  const baseline: BaselineData = existingBaseline || {
+    version: BASELINE_VERSION,
+    lastUpdated: now,
+    components: {},
+    history: [],
+  };
```

Comment:
not sure how important it is to avoid mutating the existingBaseline. If not important, skip this suggestion.

If important, you could maintain the original by cloning baseline to avoid mutating the original

```suggestion
  // Clone existing baseline to avoid mutating the original
  const baseline: BaselineData = existingBaseline
    ? structuredClone(existingBaseline)
    : {
        version: BASELINE_VERSION,
        lastUpdated: now,
        components: {},
        history: [],
      };

  // ... rest of the function remains the same
```